File: Data Structures & Algorithms/invert-a-binary-tree/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode invertTree(TreeNode root) {
        //case 1: no children -> do nothing
        //case 2: 1 child -> switch pointers
        //case 3: 2 children -> switch pointers

        //invertTree(root.left);
        //root.left = root.right;
        //root.right = oldLeft;
        if(root == null) return root;
        
        if(root.left != null) {
            invertTree(root.left);
            
        }
        if(root.right != null) {
            invertTree(root.right);
        }

        if(root.left != null && root.right != null) {
                TreeNode temp = root.left;
                root.left = root.right;
                root.right = temp;
        }
        else if(root.left != null) {
            root.right = root.left;
            root.left = null;
        }
        else if(root.right != null) {
            root.left = root.right;
            root.right = null;
        }

        return root;
    }
}

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:
        LCA = []

        def dfs(root):
            if not root:
                return
            
            if p.val < root.val and q.val < root.val:
                dfs(root.left)
            elif p.val > root.val and q.val > root.val:
                dfs(root.right)
            else:
                LCA.append(root)
        
        dfs(root)
        print(LCA)
        return LCA[-1]